content/en/docs/ops/integrations/spire/index.md

```diff
@@ -0,0 +1,411 @@
+    {{< /text >}}
+
+    Note that the workload will need the SPIFFE CSI Driver volume to access the SPIRE Agent socket. To accomplish this,
+    you can leverage the `spire` pod annotation template from the [Install Istio](#install-istio) section or add the CSI volume to
+    the deployment spec of your workload. Both of these alternatives are highlighted on the example snippet below:
+
+    {{< text yaml >}}
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+    name: sleep
+    spec:
+    replicas: 1
+    selector:
+        matchLabels:
+        app: sleep
+    template:
+        metadata:
+        labels:
+            app: sleep
+        # Injects custom sidecar template
+        annotations:
+            inject.istio.io/templates: "sidecar,spire"
+        spec:
+        terminationGracePeriodSeconds: 0
+        serviceAccountName: sleep
+        containers:
+        - name: sleep
+            image: curlimages/curl
+            command: ["/bin/sleep", "3650d"]
+            imagePullPolicy: IfNotPresent
+            volumeMounts:
+            - name: tmp
+            mountPath: /tmp
+            securityContext:
+            runAsUser: 1000
+        volumes:
+        - name: tmp
+            emptyDir: {}
+        # CSI volume
+        - name: workload-socket
+            csi:
+            driver: "csi.spiffe.io"
+    {{< /text >}}
+
+1. Get pod information:
+
+    {{< text bash >}}
+    $ SLEEP_POD=$(kubectl get pod -l app=sleep -o jsonpath="{.items[0].metadata.name}")
+    $ SLEEP_POD_UID=$(kubectl get pods $SLEEP_POD -o jsonpath='{.metadata.uid}')
+    {{< /text >}}
+
+1. Register the workload:
+
+    {{< text bash >}}
+    $ kubectl exec -n spire spire-server-0 -- \
+    /opt/spire/bin/spire-server entry create \
+        -spiffeID spiffe://example.org/ns/default/sa/sleep \
+        -parentID spiffe://example.org/ns/spire/sa/spire-agent \
+        -selector k8s:ns:default \
+        -selector k8s:pod-uid:$SLEEP_POD_UID \
+        -dns $SLEEP_POD \
+        -socketPath /run/spire/sockets/server.sock
+    {{< /text >}}
+
+{{< warning >}}
+SPIFFE IDs for workloads must follow the Istio SPIFFE ID pattern: `spiffe://<trust.domain>/ns/<namespace>/sa/<service-account>`
+{{< /warning >}}
+
+See the [SPIRE help on Registering workloads](https://spiffe.io/docs/latest/deploying/registering/) to learn how to create new entries for workloads and get them attested using multiple selectors to strengthen attestation criteria.
+
+## Verifying that identities were created for workloads
+
+Use the following command to confirm that identities were created for the workloads:
+
+{{< text bash >}}
+$ kubectl exec -i -t $SPIRE_SERVER_POD -n spire -c spire-server -- /bin/sh -c "bin/spire-server entry show -socketPath /run/spire/sockets/server.sock"
+Found 3 entries
+Entry ID         : c8dfccdc-9762-4762-80d3-5434e5388ae7
+SPIFFE ID        : spiffe://example.org/ns/istio-system/sa/istio-ingressgateway-service-account
```

Comment:
@alexandrealvino  How about the TLS communication between istiod and istio-proxy? which kind of certification is used here?

Reply:
The TLS communication between istiod and istio-agent uses a certificate generated by istiod itself, as per default behavior. That communication wasn't changed at all.

Reply:
> The TLS communication between istiod and istio-agent uses a certificate generated by istiod itself, as per default behavior. That communication wasn't changed at all.

Thanks, in fact, my question is why not integrate istiod to SPIRE also. It will look more complete. Or will it be the next step or anything block this?

Reply:
It's the next step.